docs/v0.7/en/contributor-guide/datanode/data-persistence-indexing.md

```diff
@@ -28,11 +28,54 @@ Apache Parquet file format provides inherent statistics in headers of column chu
 
 For example, in the above Parquet file, if you want to filter rows where `name` = `Emily`, you can easily skip row group 0 because the max value for `name` field is `Charlie`. This statistical information reduces IO operations.
 
-Besides Parquet's built-in statistics, our team is working on supporting a separate index file that utilizes some time-series specific indexing techniques to improve scanning performance.
+## Index Files
+
+For each SST file, in addition to the internal index within the SST file, GreptimeDB also creates a separate file to store  index structures for that SST file.
+
+The index files utilize the [Puffin][3] format, which offers significant flexibility, allowing for the storage of additional metadata and supporting a broader range of index structures.
+
+![Puffin](/puffin.png)
+
+Currently, the inverted index serves as the first separate index structure supported, stored in the index file in the form of a Blob.
+
+## Inverted Index
+
+In version 0.7, GreptimeDB introduced the inverted index to accelerate queries.
+
+The inverted index is a common index structure used for full-text searches, mapping each word in the document to a list of documents containing that word. We have applied this technology, originating from search engines, to the time series database GreptimeDB.
+
+Search engines and time series databases belong to two different domains. Although the principle of the inverted index technology applied is similar, there are some conceptual migrations:
+1. Term: In GreptimeDB, it refers to the column value of the time series.
+2. Document: In GreptimeDB, it refers to the data segment containing multiple time series.
+
+The introduction of the inverted index allows GreptimeDB to skip data segments that do not meet the conditions during queries, thereby improving scanning performance.
+
+![Inverted index searching](/inverted-index-searching.png)
+
+For example, in the above query, retrieve the data segments where `job` = `apiserver`, `handler` matches the regex `.*users`, and `status` matches the regex `4..` from the inverted index, and get the data segment list `[1]`. Subsequently, only the data segment needs to be scanned, reducing a large number of IO operations.
+
+### Inverted Index Format
+
+The inverted index is stored by column internally, and each column's inverted index block consists of FST and a Bitmap storing data segment IDs.
```

Comment:
This description confuses user, but I don't know how to modify it...

Reply:
I rephrased it, please take a look.